### ISSUE TITLE
Does subtle val-Objectness increase matter when val-Calssification keeps decrease?

### ISSUE THREAD

**Issue by Yu-Hang** [ultralytics/yolov5#3756]
## ❔Question

Hi there,

As you can see below, I'm training with 700 epochs. While the val-Objectness loss slightly increases during training, the val-Classification loss keeps decreasing (and P&R and AP keeps increasing), which is very desirable. Can this slight overfitting be ignored? Or how do you suggest to adjust otherwise?

![results_exp5](https://user-images.githubusercontent.com/5137261/123199712-65de4c80-d47d-11eb-948e-32d9c7161595.png)

## Additional context

python train.py --img 1504 --batch 10 --epochs 700 --data '/slurm_storage/yuh5/data/thin_Pf_YOLOv5/exp4/data.yaml' --weights yolov5s.pt --name yolov5s_results  --cache  --rect

lr0: 0.005
lrf: 0.2
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 0.05
cls: 0.5
cls_pw: 1.0
obj: 1.0
obj_pw: 1.0
iou_t: 0.2
anchor_t: 4.0
fl_gamma: 0.0
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0

**Comment by glenn-jocher** [comment]
@Yu-Hang well this is part of the balancing act that the 3 loss hyps play. Since your cls loss is still dropping at the end of training this indicates you have not trained long enough, I would retrain for 1k or 2k epochs until val cls begins to overfit.

In your particular case mAP is still increasing so the cls drop seems to more than offset the obj increase.

You can also adjust your cls and obj hyps:
cls: 0.5 <-- maybe increase
cls_pw: 1.0
obj: 1.0 < -- maybe decrease
obj_pw: 1.0

**Comment by Yu-Hang** [comment]
what is loss component gains? Are they the λ in the loss function?

**Comment by glenn-jocher** [comment]
@Yu-Hang it's pretty simple, you just multiply the loss by a number.

**Comment by Yu-Hang** [comment]
Hi @glenn-jocher, thanks for your suggestions. About hyp evolving, do you think in my case, a 10 epoch base scenario is sufficient? How would you decide the epoch for the base scenario?

**Comment by glenn-jocher** [comment]
@Yu-Hang the question is meaningless without context.

Evolution will create hyps evolved for best results on your base scenario. Your base scenario is up to you, not me.
